src/Illuminate/Bus/BatchRepository.php

```diff
@@ -89,4 +89,12 @@ public function delete(string $batchId);
      * @return mixed
      */
     public function transaction(Closure $callback);
+
+    /**
+     * Prune all of the entries older than the given date.
+     *
+     * @param  \DateTimeInterface  $before
+     * @return int
+     */
+    public function prune(\DateTimeInterface $before);
```

Comment:
We can't add new methods to contracts on a stable release.

Reply:
Should this be changed to merge into master (9.x)? Or do you think I should move the prune function into the command and get the connection/table through the config?

Reply:
We can already implement this with a `method_exists` call in the command which checks if `$this->laravel[BatchRepository::class]` has the method implemented or not. This can then be refactored later on if needed.

Reply:
This change has been pushed with the other tweaks.